code/modules/events/tram_malfunction.dm

```diff
@@ -0,0 +1,77 @@
+#define TRAM_MALFUNCTION_TIME_UPPER 420
+#define TRAM_MALFUNCTION_TIME_LOWER 240
+#define XING_STATE_GREEN 0
+
+/datum/round_event_control/tram_malfunction
+	name = "Tram Malfunction"
+	typepath = /datum/round_event/tram_malfunction
+	weight = 20
+	max_occurrences = 4
+	category = EVENT_CATEGORY_ENGINEERING
+
+// If manually triggered, check if there's a tram.
+/datum/round_event_control/tram_malfunction/admin_setup()
+	for(var/tram_id in GLOB.active_lifts_by_type)
+		var/datum/lift_master/tram_ref = GLOB.active_lifts_by_type[tram_id][1]
+		if(!tram_ref.specific_lift_id == MAIN_STATION_TRAM)
```

Comment:
This check will never pass. `!tram_ref.specific_lift_id` will execute first, which will then check `FALSE == MAIN_STATION_TRAM`.

You should not even need this check, your next if check is good enough.

Reply:
Yeah, made this check a lot tidier, no more duplication.